src/broadcast/broadcastmanager.cpp

```diff
@@ -6,16 +6,20 @@
 #include "engine/sidechain/enginesidechain.h"
 #include "soundio/soundmanager.h"
 
-BroadcastManager::BroadcastManager(UserSettingsPointer pConfig,
+BroadcastManager::BroadcastManager(SettingsManager* pSettingsManager,
                                    SoundManager* pSoundManager)
-        : m_pConfig(pConfig) {
+        : m_pConfig(pSettingsManager->settings()) {
     QSharedPointer<EngineNetworkStream> pNetworkStream =
             pSoundManager->getNetworkStream();
+
     if (!pNetworkStream.isNull()) {
         m_pBroadcast = QSharedPointer<EngineBroadcast>(
```

Comment:
Hmm yes, absolutely. I got used to shared pointers because of convenience but in this case a unique pointer is a better choice.